roles/azure_manage_networking_stack/tasks/delete.yml

```diff
@@ -1,25 +1,41 @@
 ---
-- name: Delete resource group
-  ansible.builtin.import_role:
-    name: cloud.azure_ops.azure_manage_resource_group
-  vars:
-    azure_manage_resource_group_operation: delete
-    azure_manage_resource_group_name: "{{ azure_manage_networking_stack_resource_group }}"
-    azure_manage_resource_group_force_delete_nonempty: true
-    azure_manage_resource_group_force_delete_locked: true
-  when: azure_manage_networking_stack_delete_option == 'all'
+- name: Check delete option
+  ansible.builtin.fail:
+    msg: Please provide delete_option as one of 'all', 'virtual_network', 'subnet'
+  when: azure_manage_networking_stack_delete_option not in ['all', 'virtual_network', 'subnet']
+
+- name: Check virtual network
```

Comment:
Can be a situation when a virtual network is already deleted but the networking stack still exists. For example when deletion was done with 'virtual_network' option and after then with 'all' option. I think Check virtual network and Check subnet tasks are redundant.

Reply:
works as expected. No need to change